### ISSUE TITLE
Semicolon check is ignored when init plain list/object

### ISSUE THREAD

**Issue by nnabinh** [palantir/tslint#2932]
### Bug Report

- __TSLint version__: ~4.5.1
- __TypeScript version__: ~2.2.2
- __Running TSLint via__: Visual Studio & Emacs tide

#### TypeScript code being linted

```ts
const executedCodes: string[] = []
```

with `tslint.json` configuration:

```json
"semicolon": [true, "always"]
```

#### Actual behavior
No warning/error.

#### Expected behavior
Show warning/error.

**Comment by JoshuaKGoldberg** [comment]
I can't reproduce this either. Resolving as aged away.
